keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -514,310 +481,126 @@ spec:
+reconnected to kubelet (kubelet restart). It also ensures that drivers which
+don't get started up again at all don't leave stale `ResourceSlices`
+behind. Garbage collection does not help in this case because the node object
+still exists. For the same reasons, the ResourceSlices belonging to a driver
+get removed when the driver unregisters, this time with a field filter for node
+name and driver name.
+
+Deleting `ResourceSlices` is possible because all information in them can be
+reconstructed by the driver. This has no effect on already allocated claims
+because the allocation result is tracked in those claims, not the
+`ResourceSlice` objects (see [below](#state-and-communication)).
+
+Embedded inside each `ResourceSlice` is the representation of one or more device.
 
 ```yaml
 kind: ResourceSlice
-apiVersion: resource.k8s.io/v1alpha2
+apiVersion: resource.k8s.io/v1alpha3
 ...
 spec:
   nodeName: worker-1
   driverName: cards.dra.example.com
-  namedResources:
-  - name: gpu-0
+  devices:
+  - name: card-1
     attributes:
-    - name: UUID
-      string: GPU-ceea231c-4257-7af7-6726-efcb8fc2ace9
-    - name: driverVersion
-      string: 1.2.3
-    - name: runtimeVersion
-      string: 11.1.42
-    - name: memory
-      quantity: 16Gi
+    - name: manufacturer # a vendor-specific attribute, automatically qualified as cards.dra.example.com/manufacturer
+      string: Cards-are-us Inc.
     - name: productName:
-      string: ACME T1000 32GB
-  - name: gpu-1
-    attributes:
-    - name: UUID
-      string: GPU-6aa0af9e-a2be-88c8-d2b3-2240d25318d7
+      string: ACME T1000 16GB
     - name: driverVersion
-      string: 1.2.3
+      version: 1.2.3
     - name: runtimeVersion
       string: 11.1.42
     - name: memory
-      quantity: 32Gi
-    - name: productName:
-      string: ACME A4-PCIE-40GB
+      quantity: 16Gi
+    - name: powerSavingSupported
+      bool: true
+    - name: dra.k8s.io/pciRoot # a fictional standardized attribute, not actually part of this KEP
+      string: pci-root-0
 ```
 
-Where "gpu-0" represents one type of card and "gpu-1" represents another (with
-the attributes hanging off each serving to "define" their individual
-properties).
-
-Compared to labels, attributes in this model have values of exactly one type. As
+Compared to labels, attributes have values of exactly one type. As
 described later on, these attributes can be used in CEL expressions to select a
 specific resource for allocation on a node.
 
-While this model is still hypothetical, we do imagine real-world models
-attaching attributes to their resources in a similar way. To avoid any future
-conflicts, we plan to reserve any attributes with the ".k8s.io" suffix for
-future use and standardization by Kubernetes. This could be used to describe
+To avoid any future conflicts, we reserve any attributes with the ".k8s.io" domain suffix
+for future use and standardization by Kubernetes. This could be used to describe
```

Comment:
I haven't been following all the latest on this, but isn't it now a `k8s.io/` prefix?

Reply:
Yes. Will change.